support/agent/miner_state.go

```diff
@@ -0,0 +1,242 @@
+package agent
```

Comment:
it is rather unfortunate that we have to create these shims inside actors, given that we've done the same thing inside lotus...

Reply:
(just a note for future thinking)